### DISCUSSION TITLE
useQueue and useEffect

### DISCUSSION THREAD

**Discussion by aurelien-boubennec** [mantinedev/mantine#5429] [Q&A]
Hi,

I was wondering if I was doing it right:

`React.useEffect(() => {
        if (myQueue.queue.length > 0) {
            //LOGIC
            myQueue.cleanQueue();
        }
    }, [myQueue.state]);`

The issue is that myQueue.state does not seem to trigger the useeffect when I add elements to the queue.
Does the code seem correct? Thanks.

**Comment by rtivital** [comment]
You are subscribing to `myQueue.state` while operating with `myQueue.queue`. If you want to watch `myQueue.queue` changes, then add it to useEffect dependencies array.
